web/templates/company/form.html.eex

```diff
@@ -6,31 +6,96 @@
+      <div class="row">
+        <div class="col-sm-6">
+          <%= text_input f, :phone, class: "form-control", placeholder: "Phone (optional)" %>
+          <%= error_tag f, :phone %>
+        </div>
+        <div class="col-sm-6">
+          <%= text_input f, :email, class: "form-control", placeholder: "Email (optional)" %>
+          <%= error_tag f, :phone %>
+        </div>
+      </div>
+    </div>
 
-  <div class="form-group">
-    <%= label f, :email, class: "control-label" %>
-    <%= text_input f, :email, class: "form-control" %>
-    <%= error_tag f, :email %>
-  </div>
+    <div class="form-group">
+      <%= text_input f, :website_url, class: "form-control", placeholder: "Business website (optional)" %>
+      <%= error_tag f, :website_url %>
+    </div>
 
-  <div class="form-group">
-    <%= label f, :description, class: "control-label" %>
-    <%= text_input f, :description, class: "form-control" %>
-    <%= error_tag f, :description %>
-  </div>
+    <div class="form-group">
+      <div class="row">
+        <div class="col-sm-6">
+          <%= text_input f, :facebook, class: "form-control", placeholder: "Facebook page (optional)" %>
+          <%= error_tag f, :facebook %>
+        </div>
+        <div class="col-sm-6">
+          <%= text_input f, :twitter, class: "form-control", placeholder: "Twitter (optional)" %>
+          <%= error_tag f, :twitter %>
+        </div>
+      </div>
+    </div>
+  </fieldset>
+
+  <fieldset class="form-group">
+    <h4>Business Location</h4>
+
+    <div class="form-group">
+      <div class="row">
+        <div class="col-sm-6">
+          <%= text_input f, :location_name, class: "form-control", placeholder: "Location name (optional)" %>
+          <%= error_tag f, :website_url %>
+        </div>
+        <div class="col-sm-6">
+          <%= text_input f, :location_street, class: "form-control", placeholder: "Location street (optional)" %>
+          <%= error_tag f, :location_street %>
+        </div>
+      </div>
+    </div>
+
+    <div class="form-group">
+      <div class="row">
+        <div class="col-sm-4">
+          <%= text_input f, :location_city, class: "form-control", placeholder: "Location city (optional)" %>
+          <%= error_tag f, :location_city %>
+        </div>
+        <div class="col-sm-2">
+          <%= text_input f, :location_zip, class: "form-control", placeholder: "Zip code" %>
+          <%= error_tag f, :location_zip %>
+        </div>
+        <div class="col-sm-6">
+          <%= select f, :location_country, @countries, class: "form-control", prompt: "Choose your country" %>
+          <%= error_tag f, :location_country %>
+        </div>
+      </div>
+    </div>
+  </fieldset>
+
+  <fieldset class="form-group">
+    <h4>Opening Hours</h4>
+    <div class="form-group">
+      <textarea class="form-control" id="company_opening_hours" name="company[opening_hours]"><%= Poison.encode!(@changeset.model.opening_hours) %></textarea>
```

Comment:
Given that the `opening_hours` field in the database is valid, that shouldn't happen - right? If for some reason it does a steypa Elixir will probably catch it and return a 500 error.